tests/conftest.py

```diff
@@ -49,11 +60,12 @@ def pytest_itemcollected(item):
     # or class name, and the docstring of the test case, then set the nodeid
     # so that pytest-spec will correctly parse the information.
     path = os.path.relpath(str(item.fspath))
-    prefix = parent.__doc__ or getattr(parent, '__name__', parent.__class__.__name__)
-    suffix = node.__doc__.strip() if node.__doc__ else node.__name__
+    prefix = docline(parent) or getattr(parent, '__name__', parent.__class__.__name__)
+    suffix = docline(node) or node.__name__
 
     # Add parametrize or test generation id to distinguish it in output
-    if item._genid:
+    # TODO: this is broken with pytest 4.2 (no attribute _genid)
+    if hasattr(item, "_genid") and item._genid:
         suffix += " ({})".format(item._genid)
```

Comment:
This fixes the broken tests but we no longer get good output for our parametrize tests. E.g what use to be:

```

Test the dataset loading functions
    [PASS]  Test loading the concrete regression dataset
    [PASS]  Test loading the energy multi regression dataset
    [PASS]  Test loading the credit binary classification dataset
    [PASS]  Test loading the occupancy binary classification dataset
    [PASS]  Test loading the mushroom binary classification dataset
    [PASS]  Test loading the hobbies text corpus dataset
    [PASS]  Test loading the game multiclass classification dataset
    [PASS]  Test loading the bikeshare regression dataset
    [PASS]  Test loading the spam binary classification dataset
    [PASS]  Test loading the walking activity clustering dataset
    [PASS]  Test loading datasets as pandas objects (load bikeshare)
    [PASS]  Test loading datasets as pandas objects (load concrete)
    [PASS]  Test loading datasets as pandas objects (load credit)
    [PASS]  Test loading datasets as pandas objects (load energy)
    [PASS]  Test loading datasets as pandas objects (load game)
    [PASS]  Test loading datasets as pandas objects (load mushroom)
    [PASS]  Test loading datasets as pandas objects (load occupancy)
    [PASS]  Test loading datasets as pandas objects (load spam)
    [PASS]  Test loading datasets as pandas objects (load walking)
    [PASS]  Test loading datasets as numpy defaults (load bikeshare)
    [PASS]  Test loading datasets as numpy defaults (load concrete)
    [PASS]  Test loading datasets as numpy defaults (load credit)
    [PASS]  Test loading datasets as numpy defaults (load energy)
    [PASS]  Test loading datasets as numpy defaults (load game)
    [PASS]  Test loading datasets as numpy defaults (load mushroom)
    [PASS]  Test loading datasets as numpy defaults (load occupancy)
    [PASS]  Test loading datasets as numpy defaults (load spam)
    [PASS]  Test loading datasets as numpy defaults (load walking)
```

Is now:

```
Test the dataset loading functions (tests/test_datasets/test_loaders.py)
    [PASS]  Test loading the concrete regression dataset
    [PASS]  Test loading the energy multi regression dataset
    [PASS]  Test loading the credit binary classification dataset
    [PASS]  Test loading the occupancy binary classification dataset
    [PASS]  Test loading the mushroom binary classification dataset
    [PASS]  Test loading the hobbies text corpus dataset
    [PASS]  Test loading the game multiclass classification dataset
    [PASS]  Test loading the bikeshare regression dataset
    [PASS]  Test loading the spam binary classification dataset
    [PASS]  Test loading the walking activity clustering dataset
    [PASS]  Test loading datasets as pandas objects
    [PASS]  Test loading datasets as pandas objects
    [PASS]  Test loading datasets as pandas objects
    [PASS]  Test loading datasets as pandas objects
    [PASS]  Test loading datasets as pandas objects
    [PASS]  Test loading datasets as pandas objects
    [PASS]  Test loading datasets as pandas objects
    [PASS]  Test loading datasets as pandas objects
    [PASS]  Test loading datasets as pandas objects
    [PASS]  Test loading datasets as numpy defaults
    [PASS]  Test loading datasets as numpy defaults
    [PASS]  Test loading datasets as numpy defaults
    [PASS]  Test loading datasets as numpy defaults
    [PASS]  Test loading datasets as numpy defaults
    [PASS]  Test loading datasets as numpy defaults
    [PASS]  Test loading datasets as numpy defaults
    [PASS]  Test loading datasets as numpy defaults
    [PASS]  Test loading datasets as numpy defaults
```

(Although with spec and verbosity turned off, this doesn't display either).